### PR TITLE
Improve error message by not displaying the first part of the second …

### LINKED ISSUE #12003: [BUU] Improve error message when no products are changed
## Description
**- As a:** enterprise user or super admin
**- On page:** `/admin/products` under feature toggle `admin_style_v3`
**- I want to be able to do:** 

With the new product page, if there is an error and no products are changed, this is the error the user sees:

image

The proposal is to improve this message by not displaying the first part of the second sentence if no changes were counted.

 The message would then be:

0 products modified.
1 product could not be saved. Please review the errors and try again.

## Acceptance Criteria & Tests

1. Activate admin_style_v3
2. Go to `admin/products`
3. Remove a price (or another mandatory info) and try to save
4. See the error message has changed.
5. Repeat the test but add a change that does work on another product: see the error message counts 1 product that was saved correctly

### PR DESCRIPTION
**Pull request by isidzukuri**
#### What? Why?
Improve error message when no products are changed

- Closes #12003

With the new product page, if there is an error and no products are changed, this is the error the user sees:

image

The proposal is to improve this message by not displaying the first part of the second sentence if no changes were counted.

The message would then be:

0 products modified.
1 product could not be saved. Please review the errors and try again.

#### What should we test?
1. Activate admin_style_v3
2. Go to admin/products
3. Remove a price (or another mandatory info) and try to save
4. See the error message has changed.
5. Repeat the test but add a change that does work on another product: see the error message counts 1 product that was saved correctly

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

### CODE REVIEW

**Comment by sigmundpetersen** [comment]
Thank you @isidzukuri !
Is this ready for code review? It's marked as draft

**Comment by isidzukuri** [comment]
@sigmundpetersen i wasn't able to test it locally and attach to the PR screenshot of UI after my changes. I need to spend more time yo understand how system works. But according to rspec my update works as expected. So if its enough for you i don't mind about review and merge. Feel free to convert it from draft to PR

**Comment by sigmundpetersen** [comment]
Yes, let's convert it and get a dev to review.
In the meantime you can get more familiar with the system. Don't hesitate to ask if there is anything!
We're also on [slack](https://join.slack.com/t/openfoodnetwork/shared_invite/zt-9sjkjdlu-r02kUMP1zbrTgUhZhYPF~A) :)

**Comment by isidzukuri** [comment]
@sigmundpetersen cool! thank you!

**Review by mkllnk — APPROVED** [review]
Thank you!

Just a little process note: we prefer a rebase on master over merging master the branch. It keeps the Git history cleaner and makes sure that any conflict resolutions are persisted.

**Comment by isidzukuri** [comment]
ok, got it :+1:

File: spec/system/admin/products_v3/products_spec.rb [hunk 1]
```diff
@@ -330,6 +330,34 @@
     end
   end
 
+  context "when only one product edited with invalid data" do
```
Comment (dacook) [inline spec/system/admin/products_v3/products_spec.rb]: I think this block belongs under the "updating"->"with invalid data" section

Comment (dacook) [reply spec/system/admin/products_v3/products_spec.rb]: Ah, you did move it later :)

**Review by dacook — APPROVED** [review]
Great, thanks for your contribution!

**Comment by RachL** [comment]
Many thanks for your contribution @isidzukuri ! This is working as expected 🤗

**Pull request merged**